Generic DataLayer and Testing

ncostar opened the pull request.

ncostar committed:
Remove Person Test

ncostar committed:
Fix + Name check

ncostar committed:
CodeCamp Updates

Person attributes, create person

ncostar committed:
Remove Person + Tests

- Merged Create and Remove tests into one using Guard Assertion convention.
- Add Etag property to Person model.
- Tests green

ncostar committed:
Person class consistent with earlier example

ncostar committed:
Get and Update - CodeCamp

ncostar committed:
Added Remove 7 Update Methods + Tests

ncostar committed:
Replace concurrency example with Update method

ncostar committed:
Remove method refactor

ncostar committed:
CodeCamp Generic Refactor + Added Interface

DanielLarsenNZ commented on DataLayer/DataLayer.cs:
```diff
@@ -0,0 +1,79 @@
+        /// <summary>
+        /// Method taking the response object and parsing it into a item with the ETag property extracted.
+        /// </summary>
+        private T ItemWithETag(ItemResponse<T> response)
+        {
+            T item = response.Resource;
+            item.ETag = response.ETag;
+            return item;
+        }
```
Actually, I think we can just decorate the ETag property with a JsonProperty. See the Model class to see what I mean.

DanielLarsenNZ commented on DataLayer/Model.cs:
```diff
@@ -0,0 +1,17 @@
+        [JsonProperty("id")]
+        public string Id { get; set; }
+
+        public string ETag { get; set; }
```
Add

```csharp
[JsonProperty("_etag")]
```

and this should serialise / de-serialise from Cosmos without the need for an ItemWithETag helper

DanielLarsenNZ commented on DataLayer/Person.cs:
```diff
@@ -1,4 +1,5 @@
-﻿using System;
```
Person.cs should be moved to the Test project and deleted from this project as per #2

DanielLarsenNZ commented on DataLayer/MockPeopleData.cs:
```diff
@@ -0,0 +1,36 @@
+﻿using System;
```
MockPeopleData can be deleted

DanielLarsenNZ commented on DataLayer/Pet.cs:
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Pet.cs can be deleted :)

ncostar committed:
Codecamp amendments
- Deleted MockPeopleData.cs
- Removed eTag parsing helper method
- Person Object Removal + Etag fix

Fixes #2 and #3

ncostar commented:
Should be resolved now @DanielLarsenNZ
